packages/engine/README.md

```diff
@@ -229,6 +229,26 @@ In order to see more logging information while the simulation is running, you ca
 export RUST_LOG=debug
 ```
 
+If your simulation takes a lot of memory and is using JavaScript behaviors, the JavaScript runner might run out of memory.
+The first step you can take is passing a larger heap size to the runner:
+
+```shell
+cargo run --bin cli -- --project /path/to/my-hash-project single-run --num-steps <NUM-STEPS> --js-runner-max-heap-size <NEW-SIZE-IN-MB>
+```
+
+This will increase the heap size but you may still run into limitations.
+The next step is to recompile the underlying JavaScript engine and setting flags for it:
+
+```shell
+export V8_FROM_SOURCE = "1"
+export GN_ARGS = "v8_enable_pointer_compression=false v8_enable_shared_ro_heap=true"
+```
+
+- `V8_FROM_SOURCE` will force the V8 engine to be compiled from source and not use a pre-compiled version.
+This will take a fairly long time, expect 15min minimum. This can be mitigated in the subsequent compilations by using [sccache](https://github.com/mozilla/sccache) or [ccache](https://ccache.dev/). Our build scripts will detect and use them. Set the `$SCCACHE` or `$CCACHE` environmental variable if it's not in your path.
+- `v8_enable_pointer_compression` is an optimization reducing RAM usage at the cost of max heap size.
```

Comment:
What does pointer compression specifically?

Reply:
V8 is 64bits so all pointers are 8 bytes long.
Pointer compression will cut these 8 bytes in half. The higher 4 bytes are stored in a register and are common to all pointers.
Then each pointer uses 4 bytes for the "partial address" and the remaining 4 are used to store values like small integers or tags.
When a pointer is used V8 will add the 4 higher bytes from the register to the "partial address" and get a "real memory address".

Reply:
Can you add a description to the readme?

Reply:
Is it useful? This is going very deep into how V8 works.

Reply:
I don't think we need the whole description, but we want more than just "reduces RAM, increases max heap size". Something like

```
- `v8_enable_pointer_compression` is an optimization reducing RAM usage at the cost of max heap size.
  Disabling this will allow addressing more than 4 gigabytes of RAM.
```

Reply:
I think that's fine, thank you!